xcode/Subconscious/Shared/Components/AppView.swift

```diff
@@ -2403,11 +2411,13 @@ struct AppModel: ModelProtocol {
     static func presentRecoveryMode(
         state: Self,
         environment: Environment,
-        presented: Bool
+        isPresented: Bool
     ) -> Update<Self> {
-        return RecoveryModeCursor.update(
-            state: state,
-            action: .presented(presented),
+        var model = state
+        model.isRecoveryModePresented = isPresented
+        return update(
+            state: model,
+            action: .recoveryMode(.requestPresent(isPresented)),
```

Comment:
Pass down action to child component. Right now this is a no-op, but it's good hygiene in case we end up wanting to do something with that action at the child level at some point.